tests/ui/asm/x86_64/issue-96797.rs

```diff
@@ -11,7 +11,14 @@ use std::arch::global_asm;
 #[no_mangle]
 fn my_func() {}
 
-global_asm!("call_foobar: jmp {}", sym foobar);
+global_asm!("
+.globl call_foobar
```

Comment:
This is the actual change that matters. The rest is more to match what rustc would produce. (a separate section for each function and proper type and size metadata for the symbol.)